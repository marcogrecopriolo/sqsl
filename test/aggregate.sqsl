{ populate source }
output to "aggrsource.res";
for  i in 10, 20, 30, 40, 50
do
        display i, i+1, i+2 format delimited;
end for;
append to default;

{ baseline }
display "input 10 to 50 - baseline";
input from "aggrsource.res" pattern delimited;

{ test and alter field, awk syntax }
display "input 10 to 50 - 31->35";
input from "aggrsource.res" pattern delimited
    aggregate ($2==31) $2+=4;

{ test and alter field, sql syntax }
display "input 10 to 50 - 42->47";
input from "aggrsource.res" pattern delimited
    aggregate when column(3)==42 then let column(3)+=5;
