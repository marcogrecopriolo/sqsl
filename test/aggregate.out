input 10 to 50 - baseline
10,11,12
20,21,22
30,31,32
40,41,42
50,51,52
input 10 to 50 - 31->35
10,11,12
20,21,22
30,35,32
40,41,42
50,51,52
input 10 to 50 - 42->47
10,11,12
20,21,22
30,31,32
40,41,47
50,51,52
